`SimpleArrayField` should accept `forms.Field` not `Type[forms.Field]`

# Bug report

## What's wrong

If you define a form with a `SimpleArrayField`, like below
```python
from django import forms
from django.contrib.postgres.forms import SimpleArrayField

class TestForm(forms.Form):
	lots_of_dates = SimpleArrayField(forms.DateField(), required=False)
```

You get the following error
```
temporarium/forms.py:117: error: Argument 1 to "SimpleArrayField" has incompatible type "DateField"; expected "Type[Field]"
```

If I pass `forms.DateField` (without the `()`) then mypy is happy, but at runtime, when submitting the form, I get
```
Traceback (most recent call last):
2022-08-10T12:11:50.152786710Z   File "/venv/lib/python3.10/site-packages/django/contrib/staticfiles/handlers.py", line 76, in __call__
2022-08-10T12:11:50.152788835Z     return self.application(environ, start_response)
2022-08-10T12:11:50.152790669Z   File "/venv/lib/python3.10/site-packages/django/core/handlers/wsgi.py", line 133, in __call__
2022-08-10T12:11:50.152792419Z     response = self.get_response(request)
2022-08-10T12:11:50.152794002Z   File "/venv/lib/python3.10/site-packages/django/core/handlers/base.py", line 130, in get_response
2022-08-10T12:11:50.152795752Z     response = self._middleware_chain(request)
2022-08-10T12:11:50.152797294Z   File "/venv/lib/python3.10/site-packages/django/core/handlers/exception.py", line 49, in inner
2022-08-10T12:11:50.152799002Z     response = response_for_exception(request, exc)
2022-08-10T12:11:50.152800585Z   File "/venv/lib/python3.10/site-packages/django/core/handlers/exception.py", line 114, in response_for_exception
2022-08-10T12:11:50.152802294Z     response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
2022-08-10T12:11:50.152804127Z   File "/venv/lib/python3.10/site-packages/django/core/handlers/exception.py", line 149, in handle_uncaught_exception
2022-08-10T12:11:50.152806002Z     return debug.technical_500_response(request, *exc_info)
2022-08-10T12:11:50.152807544Z   File "/venv/lib/python3.10/site-packages/django_extensions/management/technical_response.py", line 40, in null_technical_500_response
2022-08-10T12:11:50.152809294Z     raise exc_value.with_traceback(tb)
2022-08-10T12:11:50.152810877Z   File "/venv/lib/python3.10/site-packages/django/core/handlers/exception.py", line 47, in inner
2022-08-10T12:11:50.152812752Z     response = get_response(request)
2022-08-10T12:11:50.152814294Z   File "/venv/lib/python3.10/site-packages/django/core/handlers/base.py", line 181, in _get_response
2022-08-10T12:11:50.152816169Z     response = wrapped_callback(request, *callback_args, **callback_kwargs)
2022-08-10T12:11:50.152817752Z   File "/home/appuser/vectare/decorators.py", line 18, in wrapper
2022-08-10T12:11:50.152819544Z     return func(request, *args, **kwargs)
2022-08-10T12:11:50.152821294Z   File "/home/appuser/vectare/decorators.py", line 18, in wrapper
2022-08-10T12:11:50.152823044Z     return func(request, *args, **kwargs)
2022-08-10T12:11:50.152824794Z   File "/venv/lib/python3.10/site-packages/django/views/generic/base.py", line 70, in view
2022-08-10T12:11:50.152826460Z     return self.dispatch(request, *args, **kwargs)
2022-08-10T12:11:50.152828044Z   File "/venv/lib/python3.10/site-packages/django/views/generic/base.py", line 98, in dispatch
2022-08-10T12:11:50.152829710Z     return handler(request, *args, **kwargs)
2022-08-10T12:11:50.152841294Z   File "/venv/lib/python3.10/site-packages/django/views/generic/edit.py", line 141, in post
2022-08-10T12:11:50.152843127Z     if form.is_valid():
2022-08-10T12:11:50.152844835Z   File "/venv/lib/python3.10/site-packages/django/forms/forms.py", line 175, in is_valid
2022-08-10T12:11:50.152846460Z     return self.is_bound and not self.errors
2022-08-10T12:11:50.152848002Z   File "/venv/lib/python3.10/site-packages/django/forms/forms.py", line 170, in errors
2022-08-10T12:11:50.152849669Z     self.full_clean()
2022-08-10T12:11:50.152851169Z   File "/venv/lib/python3.10/site-packages/django/forms/forms.py", line 372, in full_clean
2022-08-10T12:11:50.152852794Z     self._clean_fields()
2022-08-10T12:11:50.152854335Z   File "/venv/lib/python3.10/site-packages/django/forms/forms.py", line 390, in _clean_fields
2022-08-10T12:11:50.152856169Z     value = field.clean(value)
2022-08-10T12:11:50.152857877Z   File "/venv/lib/python3.10/site-packages/django/contrib/postgres/forms/array.py", line 31, in clean
2022-08-10T12:11:50.152859502Z     value = super().clean(value)
2022-08-10T12:11:50.152861085Z   File "/venv/lib/python3.10/site-packages/django/forms/fields.py", line 149, in clean
2022-08-10T12:11:50.152862710Z     value = self.to_python(value)
2022-08-10T12:11:50.152864252Z   File "/venv/lib/python3.10/site-packages/django/contrib/postgres/forms/array.py", line 50, in to_python
2022-08-10T12:11:50.152865960Z     values.append(self.base_field.to_python(item))
2022-08-10T12:11:50.152867710Z TypeError: DateField.to_python() missing 1 required positional argument: 'value'
```

## How is that should be

mypy should accept the original code above

I have worked around this in my project by defining a custom type for `SimpleArrayField`, changing `base_field: Type[forms.Field]` for `forms.Field`

## System information

- OS: Debian Buster
- `python` version: 3.10
- `django` version: 3.2
- `mypy` version: 0.971
- `django-stubs` version: 1.12.0
- `django-stubs-ext` version: 0.5.0

--- a/django-stubs/contrib/postgres/forms/array.pyi
+++ b/django-stubs/contrib/postgres/forms/array.pyi
@@ -19,7 +19,7 @@ class SimpleArrayField(forms.CharField):
     max_length: Optional[int]
     def __init__(
         self,
-        base_field: Type[forms.Field],
+        base_field: forms.Field,
         *,
         delimiter: str = ...,
         max_length: Optional[int] = ...,
@@ -51,11 +51,11 @@ class SplitArrayWidget(forms.Widget):
 
 class SplitArrayField(forms.Field):
     default_error_messages: _ErrorMessagesT = ...
-    base_field: Type[forms.Field]
+    base_field: forms.Field
     size: int
     remove_trailing_nulls: bool
     def __init__(
-        self, base_field: Type[forms.Field], size: int, *, remove_trailing_nulls: bool = ..., **kwargs: Any
+        self, base_field: forms.Field, size: int, *, remove_trailing_nulls: bool = ..., **kwargs: Any
     ) -> None: ...
     def to_python(self, value: Any) -> Sequence[Any]: ...
     def clean(self, value: Any) -> Sequence[Any]: ...

--- a/django-stubs/contrib/postgres/forms/array.pyi
+++ b/django-stubs/contrib/postgres/forms/array.pyi
@@ -13,7 +13,7 @@ from ..utils import prefix_validation_error as prefix_validation_error
 
 class SimpleArrayField(forms.CharField):
     default_error_messages: _ErrorMessagesT = ...
-    base_field: Type[forms.Field]
+    base_field: forms.Field
     delimiter: str
     min_length: Optional[int]
     max_length: Optional[int]

--- a/tests/typecheck/contrib/postgres/test_fields.yml
+++ b/tests/typecheck/contrib/postgres/test_fields.yml
@@ -17,3 +17,18 @@
 
                 class MyModel(models.Model):
                     array = ArrayField(base_field=models.TextField())
+-   case: postgres_forms_simple_array_field
+    main: |
+        from myapp.forms import MyForm
+        MyForm()
+    installed_apps:
+        - myapp
+    files:
+        -   path: myapp/__init__.py
+        -   path: myapp/forms.py
+            content: |
+                from django import forms
+                from django.contrib.postgres.forms import SimpleArrayField
+
+                class MyForm(forms.Form):
+                    lots_of_dates = SimpleArrayField(forms.DateField(), required=False)

